docs/tutorial/message-ports.md

```diff
@@ -84,6 +84,80 @@ process, you can listen for the `close` event by calling `port.on('close',
 
 ## Example use cases
 
+### Setting up a MessageChannel between two renderers
+
+In this example, the main process sets up a MessageChannel, then sends each port
+to a different renderer. This allows renderers to send messages to each other
+without needing to use the main process as an in-between.
+
+```js
+// main.js ///////////////////////////////////////////////////////////////////
+const { BrowserWindow, app, MessageChannelMain } = require('electron')
+
+app.whenReady().then(async () => {
+  // create the windows.
+  const mainWindow = new BrowserWindow({
+    show: false,
+    webPreferences: {
+      contextIsolation: false,
+      nodeIntegration: true,
+      preload: 'preloadMain.js'
+    }
+  })
+
+  const secondaryWindow = BrowserWindow({
+    show: false,
+    webPreferences: {
+      contextIsolation: false,
+      nodeIntegration: true,
+      preload: 'preloadSecondary.js'
+    }
+  })
+
+  // set up the channel.
+  const { port1, port2 } = new MessageChannelMain()
+
+  // once the webContents are ready, send a port to each webContents with postMessage.
+  mainWindow.once('ready-to-show', () => {
+    mainWindow.webContents.postMessage('port', null, [port1])
+  })
+
+  secondaryWindow.once('ready-to-show', () => {
+    secondaryWindow.webContents.postMessage('port', null, [port2])
+  })
+})
+```
+
+Then, in your preload scripts you receive the port through IPC and set up the
+listeners.
+
+```js
+// preloadMain.js and preloadSecondary.js
+const { ipcRenderer } = require('electron')
+
+ipcRenderer.on('port', e => {
+  // port received, make it globally available.
+  window.messagePort = e.ports[0]
+
+  window.messagePort.onmessage = messageEvent => {
+    // handle message
+  }
+})
+```
+
+In this example messagePort is bound to the `window` object directly. It is better
+to use `contextIsolation` and set up specific contextBridge calls for each of your
+expected messages, but for the simplicity of this example we don't.
```

Comment:
I'd prefer to have a full context isolated example for best practices, but that aside: we don't need `nodeIntegration: true` in the code sample if all we're doing is attaching to `window` in the preload right?

Reply:
There's already a context isolated example further down the page: https://github.com/electron/electron/blob/7951ebb1efb615d6eb6258aaa31e59218ca2235d/docs/tutorial/message-ports.md#communicating-directly-between-the-main-process-and-the-main-world-of-a-context-isolated-page so I think for simplicity this is better: the new one shows just the setup from main to two renderers, and then the other one shows how to add context isolation, keeping it one concept per example. I'll add a reference to the next example though!

I can take the node integration out. I had it because the other examples had it too :)